providers/autodns/types.go

```diff
@@ -0,0 +1,68 @@
+package autodns
+
+import (
+	"github.com/StackExchange/dnscontrol/v3/models"
+	"github.com/StackExchange/dnscontrol/v3/providers/bind"
+)
+
+type ResourceRecord struct {
+
+	// The name of the record.
+	// Required: true
+	Name string `json:"name"`
+
+	// Preference of the record, need for some record types e.g. MX
+	// Maximum: 65535
+	Pref int32 `json:"pref,omitempty"`
+
+	// The bind notation of the record. Only used by the zone stream task!
+	Raw string `json:"raw,omitempty"`
+
+	// TTL of the record (Optionally if not set then Default SOA TTL is used)
+	TTL int64 `json:"ttl,omitempty"`
+
+	// The type of the record, e.g. A
+	// Example: A, MX, CNAME, TXT, SRV, PTR, AAAA, NS, CAA, PROCEED_MUC, TLSA, NAPTR, SSHFP, LOC, RP, HINFO, PROCEED, ALIAS, DNSKEY, NSEC, DS
```

Comment:
Is that an example or the definitive list of accepted records?  If it is the definitive list, change "Example" to "Permitted values".

Reply:
Fixed with https://github.com/StackExchange/dnscontrol/pull/1336/commits/4f8878b84cf2a624e9b2da4d654ba8cac42d8afe